src/test/scala/mesosphere/marathon/api/v2/AppsResourceTest.scala

```diff
@@ -60,6 +60,20 @@ class AppsResourceTest extends MarathonSpec with MarathonActorSupport with Match
     JsonTestHelper.assertThatJsonString(response.getEntity.asInstanceOf[String]).correspondsToJsonOf(expected)
   }
 
+  test("Create a new app fails with Validation errors for negative resources") {
+    Given("An app with negative resources")
+    val app = AppDefinition(id = PathId("/app"), cmd = Some("cmd"), versionInfo = OnlyVersion(Timestamp.zero), mem = -128, cpus = -1, instances = -1)
```

Comment:
Nit-pick: 

* not only the resources are negative, it would be great to test one thing at a time.
* it would be nice to break the line here because e.g. in github you cannot see the whole thing without scrolling.